guides/gatherers.md

```diff
@@ -14,32 +14,49 @@ Facts Gathering process in a nutshell
 fact = gatherer(argument)
 ```
 
+## A note of gatherers versioning
+
+When a gatherer includes a breaking change (e.g. the Rhai output format changes), a new version of that same gatherer will be introduced with a new
+version labeled as a suffix to the gatherers name.
+
+Example:
+
+- `systemd@v1` -> Represents the first version of the systemd gatherer
+- `systemd@v2` -> Represents the second version of the systemd gatherer
+
+Note that when writing a check, if no tag is specified (e.g. `systemd`), the latest version will be used. Because of this it is **strongly** recommended
+to always pin your checks to a specific version of a gatherer.
+
+Also note that not all changes in a released gatherer will get a new version tag. A new version tag will be released only for breaking changes,
+while non-breaking changes such as additional fields in the Rhai output will reuse the latest existing tag. This means that if you want to use a check
+that relies on a newer field introduced after an update, you will need to upgrade both wanda and the agent to the latest version.
```

Comment:
why do you need to update wanda to the latest version?

Reply:
What I meant is something like (using what eugen suggested):
To use a check that relies on a newer field introduced after an update, upgrade not only wanda but also the agent to the latest version

but after reading you, I think it creates more confusion like this... I'll remove that